### PR TITLE
Major Update (Fix Menu, Links, Missing French Pages)

### PR DESCRIPTION
**Pull request by thekodester**
Fixes many issue, not yet complete or tested. Work in progress.

### CODE REVIEW

**Comment by thekodester** [comment]
@duboisp @juleskuehn Is there anything so far with the changes you like to see different?

**Comment by thekodester** [comment]
The _'Research Folder'_ (https://github.com/wet-boew/wet-boew-documentation/tree/master/research) is quite Large, what I was think is for current time we can just disable French link.

The _'Test Case Folder'_ (https://github.com/wet-boew/wet-boew-documentation/tree/master/testcase) is full of html files is it okay to change them to `.md`?

File: _config.yml [hunk 1]
```diff
@@ -1,9 +1,9 @@
 # standard jekyll configuration
 
-title_en: "Documentation - Web Experience Toolkit&#160;(WET)"
-title_fr: "Documentation - Boîte à outils de l'expérience Web&#160;(BOEW)"
-description_en: "Documentation for the Web Experience Toolkit (WET)."
-description_fr: "Documentation pour la Boîte à outils de l'expérience Web (BOEW)."
+title_en: "Web Experience Toolkit (WET)"
+title_fr: "Boîte à outils de l'expérience Web (BOEW)"
+description_en: "Documentation for Web Experience Toolkit (WET)"
+description_fr: "Documentation for Web Experience Toolkit (WET)"
```
Comment (juleskuehn) [inline _config.yml]: Don't change this line

Comment (thekodester) [reply _config.yml]: Can you elvabourate as to why?

I as I stated above I have not yet done testing, but I converted the style guide to the same codec (GitHub pages) and this file is needed to be updated for my changes to for.

I do not speak French so if you can provide French translation I can update that line.

Comment (juleskuehn) [reply _config.yml]: There was already a suitable French translation in the line you replaced

Comment (thekodester) [reply _config.yml]: Did not notice. Thanks.

**Comment by juleskuehn** [comment]
This is a lot of changes! Thanks. I don't think reviewing the code line by line is the best way to go about this one - is there a working copy online?

I wouldn't change the test case HTML to MD. The HTML files are there to test behaviour for certain elements without WET being involved (to avoid using polyfills).

**Comment by thekodester** [comment]
Unfortunately that is the only way with Github pages. If it was coded correctly from the start a major update like this would not of been needed.

**Comment by thekodester** [comment]
A great example is https://wet-boew.github.io/wet-boew-documentation/index-fr you can see the entire page is broken.

**Comment by thekodester** [comment]
Believe all files updated. I went with disabling French link on research section.

Work to do:
- [ ] test changes make sure no broken

**Comment by thekodester** [comment]
All of my testing is now complete. All links appear to work properly.

@duboisp @juleskuehn

**Comment by thekodester** [comment]
To run locally use the following terminal command 'bundle exec jekyll serve'

See https://help.github.com/en/github/working-with-github-pages/testing-your-github-pages-site-locally-with-jekyll for more details.

**Word of warning when testing locally not all links well work properly and you may need to manually enter them (burden of local server testing).**

**Comment by duboisp** [comment]
Thanks @thekodester for all that work.

I haven't finished my review yet, but what is the rational of renaming the design decision files? The path was designed to be sort and to avoid wording duplication.

**Comment by thekodester** [comment]
@duboisp It was to make it more clear what the page was and to allow French of each page (makes it more inline with other WET pages and how they where named, but it could be shorten to `dd-one-en.md`.

**Comment by thekodester** [comment]
For transparent and it make it easier to review here a list of changes:
- Better Page Titles (short and less confusing)
- Main Menu (allows user to return to the main content of WET)
- Bread Crumb (allows the user to return to the document pages)
- Language select that directs to a French page if exist (if the existing page is English display a warning that it is not translated)
- Disable language selection for research section as a temp solution to no French pages
- Footer links updated to follow wet-boew and wet-boew-styleguide repo better
- Resolves many 404 links and added a proper WET 404 error
- Pages renamed to follow other wet-boew and wet-boew-styleguide repo better
- Gem files added for local testing
- html files where left untouched in test case as currently required for the testing purposes

**Comment by thekodester** [comment]
Gonna close this PR and reopen due to conflicts.

**Pull request closed without merging**